File: 프로그래머스/lv0/120921. 문자열 밀기/문자열 밀기.js
function solution(A, B) {
  var answer = 0;
  const stringToArray = A.split('');

  if (A === B) {
    return 0
  }

  while (true) {
    answer += 1;
    stringToArray.unshift(stringToArray.pop());
    if (stringToArray.join('') === B) {
      return answer
    } else if (answer === A.length) {
      return -1
    }
  }
}

File: 프로그래머스/lv0/120921. 문자열 밀기/문자열 밀기.py
def solution(A, B):
    answer = 0
    
    A = list(A)
    B = list(B)
    
    if A == B:
        return 0
    
    for _ in range(len(A)):
        answer += 1
        A.insert(0, A.pop())
        if A == B:
            return answer
    
    if A == B:
        return answer
    return -1
